src/data/seq/seq.lean

```diff
@@ -288,6 +289,20 @@ begin
   rw [h1, h2], apply H
 end
 
+/-- Embed a list as a sequence -/
+def of_list (l : list α) : seq α :=
+⟨list.nth l, λ n h, begin
+  rw list.nth_eq_none_iff at h ⊢,
+  exact h.trans (nat.le_succ n)
+end⟩
+
+instance coe_list : has_coe (list α) (seq α) := ⟨of_list⟩
```

Comment:
IMHO, it is strange to have both a function and a coercion. Which form is the `simp`-normal form? I would drop one of them.